### PR TITLE
Detect form submission errors #16

### PR DESCRIPTION
**Pull request by lacostej**

### CODE REVIEW

File: lib/codes/itunes_connect.rb [hunk 1]
```diff
@@ -40,6 +40,16 @@ def run(args)
 
       text_fields.first.set(number_of_codes.to_s)
       click_next
+
+      # are there any errors ?
+      errors = []
+      begin
+         errors = wait_for_elements("div[id=LCPurpleSoftwarePageWrapperErrorMessage]")
+      rescue Exception => e
+         # puts "Error: #{e.message}"
```
Comment (KrauseFx) [inline lib/codes/itunes_connect.rb]: I think we don't want to print the error message here, right? We should remove the comment then.

Comment (lacostej) [inline lib/codes/itunes_connect.rb]: On Sep 2, 2015 19:42, "Felix Krause" <notifications@github.com> wrote:
>
> In lib/codes/itunes_connect.rb:
>
> > @@ -40,6 +40,16 @@ def run(args)
> >
> >        text_fields.first.set(number_of_codes.to_s)
> >        click_next
> > +
> > +      # are there any errors ?
> > +      errors = []
> > +      begin
> > +         errors =
wait_for_elements("div[id=LCPurpleSoftwarePageWrapperErrorMessage]")
> > +      rescue Exception => e
> > +         # puts "Error: #{e.message}"
>
> I think we don't want to print the error message here, right? We should
remove the comment then.

I've noticed that there's an error message printed when it fails. Maybe
coming from wait_for_elementos.

I have to check it again.

**Comment by lacostej** [comment]
Cleaned up and made error red.

Errors are detected fine:

screen shot 2015-09-03 at 17 49 50

But I don't get what's printing the output  here
screen shot 2015-09-03 at 17 52 37

``````
[17:50:33]: ["/Users/lacostej/.rvm/gems/ruby-2.2.2/gems/codes-0.3.0/lib/codes/itunes_connect.rb:48:in `run'", "/Users/lacostej/.rvm/gems/ruby-2.2.2/gems/codes-0.3.0/lib/codes/codes_runner.rb:4:
```````

File: lib/codes/itunes_connect.rb [hunk 2]
```diff
@@ -41,6 +41,15 @@ def run(args)
...
       text_fields.first.set(number_of_codes.to_s)
       click_next
+
+      # are there any errors ?
+      errors = []
+      begin
+         errors = wait_for_elements("div[id=LCPurpleSoftwarePageWrapperErrorMessage]")
+      rescue Exception
+      end
+      raise errors.first.text.red unless errors.count == 0
```
Comment (KrauseFx) [inline lib/codes/itunes_connect.rb]: What if there are multiple errors? How about `errors.join("; ")`

Comment (lacostej) [inline lib/codes/itunes_connect.rb]: I believe we should be covered. The <div> is described here:

https://github.com/fastlane/codes/issues/16#issue-98120709

and encompasses all underlying errors. I can add HTML extract documentation to the code if you want.

**Comment by KrauseFx** [comment]
Looks good, thanks. I'll bundle up a new release now :+1:

**Comment by lacostej** [comment]
Question: do you know why there's a spurious print out ? See https://github.com/fastlane/codes/pull/20#issuecomment-137493008

**Pull request merged**